src/main/java/jenkins/scm/api/mixin/ChangeRequestSCMHead.java

```diff
@@ -20,46 +20,40 @@
  * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
+ *
  */
 
-package jenkins.scm.api;
+package jenkins.scm.api.mixin;
 
 import edu.umd.cs.findbugs.annotations.NonNull;
-import hudson.model.Action;
-import java.util.ArrayList;
-import java.util.List;
-import jenkins.scm.api.actions.ChangeRequestAction;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMSource;
+import org.kohsuke.stapler.export.Exported;
 
 /**
- * Base class for {@link SCMHead} instances that correspond to a change request.
+ * Mixin interface to identify {@link SCMHead} instances that correspond to a change request.
  *
- * @since FIXME
+ * @since 2.0
  */
-public abstract class ChangeRequestSCMHead extends SCMHead {
+public interface ChangeRequestSCMHead extends SCMHeadMixin {
+
     /**
-     * Constructor.
+     * Identifier of this change request.
+     * Expected to be unique among requests coming from a given {@link SCMSource}.
      *
-     * @param name the name.
+     * @return an ID of some kind, such as a pull request number (decimal) or a Gerrit change ID
      */
-    public ChangeRequestSCMHead(@NonNull String name) {
-        super(name);
-    }
+    @Exported
+    @NonNull
+    String getId();
 
     /**
-     * Returns the {@link ChangeRequestAction} of this {@link ChangeRequestSCMHead}.
+     * Branch to which this change would be merged or applied if it were accepted.
```

Comment:
Is there any mechanism to signal that this has changed since the last call, for example in the case of a GH PR with a modified base, or can an implementation freely return a different value from call to call?